### ISSUE TITLE
Skills are not greyed out when you don't have enough mana

### ISSUE THREAD

**Issue by shanaqui** [HabitRPG/habitica#13286]
Previously(?) as far as I and the user reporting it remember, tasks used to be greyed out when you didn't have enough mana to cast them. That is no longer the case: I have only 30 mana, but all these skills look usable:

![image](https://user-images.githubusercontent.com/8144640/118984854-d4f3ed00-b975-11eb-97f9-684e132c41ab.png)

This can be confusing for users, though the correct error message does appear.

**Comment by akmaekki** [comment]
Can I work on this?

**Comment by Alys** [comment]
@akmaekki Thank you! I've marked this as in progress for you. Post here if you have questions or run into trouble.

**Comment by akmaekki** [comment]
@shanaqui 
I checked this feature. It is located in the file drawer.vue (https://github.com/HabitRPG/habitica/blob/develop/website/client/src/components/ui/drawer.vue).

As far as I can tell, there was never a greying out of the skill tiles, contrary to the smartphone version. E.g. on my Android phone the tiles are greyed out in case of insufficient mana. Maybe you confused the behaviour of desktop and app version? Or maybe I am wrong. I also checked out a previous version of this file, the version when there was the migration from pug to html. However, the behaviour was the same.

So, to summarize, if there is insufficient mana:

- **Desktop version:** A message is shown "Not enough mana"
- **App version:** The tile(s) is/are greyed out and a message is shown "Not enough mana"

But we can try to make the desktop version behave the same as the app version, if wished.

**Comment by akmaekki** [comment]
@shanaqui 
I checked this feature. It is located in the file drawer.vue (https://github.com/HabitRPG/habitica/blob/develop/website/client/src/components/tasks/spells.vue).

As far as I can tell, there was never (?) a greying out of the skill tiles, contrary to the smartphone version. E.g. on my Android phone the tiles are greyed out in case of insufficient mana. Maybe you confused the behaviour of desktop and app version? Or maybe I am wrong. I also checked out a previous version of this file, the version when there was the migration from pug to html. However, the behaviour was the same. Right now it is like that, that there are some checks if spells can be used at all. Then a function `castStart` is called in there it is checked if the user has enough mana. If not, the message "Not enough mana" is displayed.

So, to summarize, if there is insufficient mana:

- **Desktop version:** A message is shown "Not enough mana"
- **App version:** The tile(s) is/are greyed out and a message is shown "Not enough mana"

But we can try to make the desktop version behave the same as the app version, if wished.

**Comment by citrusella** [comment]
Tiles are greyed out in the following two cases, at least (and I know for sure this happens on web): A) If casting again will have no effect (i.e. having cast Chilling Frost, or casting Stealth enough that it covers all remaining dailies), and B) if you have not reached the level at which you unlock the skill (i.e. if you are looking at a level 13 skill and you're level 11). Here is the code that dictates if a skill is greyed out for reasons other than level:

https://github.com/HabitRPG/habitica/blob/bc730a0d59676aabaa8bbd55fb198995b7c8ba3a/website/client/src/components/tasks/spells.vue#L465-L480

(The code that actually applies the grey-out styling does take level into account:

https://github.com/HabitRPG/habitica/blob/bc730a0d59676aabaa8bbd55fb198995b7c8ba3a/website/client/src/components/tasks/spells.vue#L69-L72)

I think you're right about the case of insufficient mana not previously having a grey-out behavior, though? To my memory? At least not on the new design? 🤔 (I'm no arbiter of what to do but I'd *personally* be thrown off by skills greyed out in that particular manner, at least for classes like healer where the skills aren't in order of mana required. But OTOH, I have a very picky brain, though, and it probably makes more *sense* for them to grey out in that situation--being thrown off because not grey-grey-grey-not grey is way less of a usability problem than not realizing a skill cast won't work before you cast it.)

**Comment by akmaekki** [comment]
@citrusella 
Thanks for your very helpful input. I didn't test the cases A and B mentioned by you, but just the case of not sufficient mana. 

Anyway, I will take another look at the previous changes to `spell.vue`. I also tried to compare with Habitica's Android repository, but don't have any experience with mobile development. If can't find a suspicuous previous change, then I would just add this functionality to grey out the respective tiles.

But on the other hand, if I change it, maybe somebody issues a bug because tiles are now greyed out on Web :-)

**Comment by shanaqui** [comment]
I think this might be a question for the designers, then -- @Tressley, tagging you in to check on what we think the desired behaviour is here. Personally, I find it odd that we have some situations where we grey out skills if they can't be used, but we aren't consistent about it.

**Comment by Tressley** [comment]
Yes, the skill should be disabled (grayed out) if the user cannot cast it.

**Comment by beffymaroo** [comment]
@akmaekki Are you still working on this issue? Please let me know, and I can adjust labels if needed. Thank you!

**Comment by codexdelta** [comment]
Can I pick this up?

**Comment by SabreCat** [comment]
@codexdelta Sure, have a look with our thanks!

**Comment by SabreCat** [comment]
@codexdelta Are you still working on this? Let us know what questions have come up!

**Comment by codexdelta** [comment]
@SabreCat will be trying to deliver this by this weekend, thanks.

**Comment by CuriousMagpie** [comment]
@codexdelta Are you still working on this? Please let us know!

**Comment by jasonmishi** [comment]
Can I work on this issue please?

**Comment by CuriousMagpie** [comment]
I've assigned this to you, @JasonCarvalho-tech! Thanks!
